vm/src/obj/objdict.rs

```diff
@@ -146,46 +149,65 @@ impl PyDict {
         !self.entries.is_empty()
     }
 
-    fn inner_eq(zelf: PyRef<Self>, other: &PyDict, vm: &VirtualMachine) -> PyResult<bool> {
-        if other.entries.len() != zelf.entries.len() {
-            return Ok(false);
+    fn inner_cmp(
+        zelf: PyRef<Self>,
+        other: PyDictRef,
+        size_func: fn(usize, usize) -> bool,
+        item: bool,
+        vm: &VirtualMachine,
+    ) -> PyResult<PyComparisonValue> {
+        if size_func(zelf.len(), other.len()) {
```

Comment:
For eq and ne, this test leads the loop comparison.
But for other comparisons like lt, gt, the size comparison doesn't guarantee subset.

Assume `a =  {'a': 1, 'b' 2}; b = {'a': 1, 'b': 3, 'c': 4}`,
`a.items() > b.items()` is false but also `b.items() > a.items()` is false.

Reply:
Thank you for your advice! but don't the cpython also return false and false? I put it at first because it seems cpython check a length at first. [link](https://github.com/python/cpython/blob/c75330605d4795850ec74fdc4d69aa5d92f76c00/Objects/dictobject.c#L4239) Should I change it as you said?

Reply:
Oh, I am sorry. I thought something totally wrong. Yes, it seems current behaviour is exactly as expected and also same as also what I described. Sorry for making confusion!

Reply:
No problem! I really want to thank you for your help.